examples/ex23.cpp

```diff
@@ -0,0 +1,988 @@
+      if (prob == beam && dim == 3) { keys = "keys macFFiYYYYYYYYYYYYYYYYYY\n"; }
+      if (prob == beam && dim == 2) { keys = "keys amrRljcUUuuu\n"; }
+
+      char vishost[] = "localhost";
+      int visport = 19916;
+
+      socketstream sol_sock_re(vishost, visport);
+      sol_sock_re.precision(8);
+      sol_sock_re << "solution\n"
+                  << *mesh << x.real() << keys
+                  << "window_title 'Solution real part'" << flush;
+
+      socketstream sol_sock_im(vishost, visport);
+      sol_sock_im.precision(8);
+      sol_sock_im << "solution\n"
+                  << *mesh << x.imag() << keys
+                  << "window_title 'Solution imag part'" << flush;
+
+      GridFunction x_t(fespace);
+      x_t = x.real();
+      socketstream sol_sock(vishost, visport);
+      sol_sock.precision(8);
+      sol_sock << "solution\n"
+               << *mesh << x_t << keys << "autoscale off\n"
+               << "window_title 'Harmonic Solution (t = 0.0 T)'"
+               << "pause\n" << flush;
+      cout << "GLVis visualization paused."
+           << " Press space (in the GLVis window) to resume it.\n";
+      int num_frames = 32;
+      int i = 0;
+      while (sol_sock)
+      {
+         double t = (double)(i % num_frames) / num_frames;
+         ostringstream oss;
+         oss << "Harmonic Solution (t = " << t << " T)";
+
+         add(cos(2.0 * M_PI * t), x.real(),
+             sin(2.0 * M_PI * t), x.imag(), x_t);
+         sol_sock << "solution\n"
+                  << *mesh << x_t
+                  << "window_title '" << oss.str() << "'" << flush;
+         i++;
+      }
+   }
+
+   // 18. Free the used memory.
+   delete A;
+   delete pml;
+   delete fespace;
+   delete fec;
+   delete mesh;
+   return 0;
+}
+
+void source(const Vector &x, Vector &f)
+{
+   Vector center(dim);
+   double r = 0.0;
+   for (int i = 0; i < dim; ++i)
+   {
+      center(i) = 0.5 * (comp_domain_bdr(i, 0) + comp_domain_bdr(i, 1));
+      r += pow(x[i] - center[i], 2.);
+   }
+   double n = 5.0 * omega * sqrt(epsilon * mu) / M_PI;
+   double coeff = pow(n, 2) / M_PI;
+   double alpha = -pow(n, 2) * r;
+   f = 0.0;
+   f[0] = coeff * exp(alpha);
+}
+
+void maxwell_solution(const Vector &x, std::vector<std::complex<double>> &E)
+{
+   // Initialize
+   for (int i = 0; i < dim; ++i)
+   {
+      E[i] = 0.0;
+   }
+
+   std::complex<double> zi = std::complex<double>(0., 1.);
+   double k = omega * sqrt(epsilon * mu);
+   switch (prob)
+   {
+      case scatter:
+      case lshape:
+      case fichera:
+      {
+         Vector shift(dim);
+         shift = 0.0;
+         if (prob == fichera) { shift = 1.0; }
+
+         if (dim == 2)
+         {
+            double x0 = x(0) + shift(0);
+            double x1 = x(1) + shift(1);
+            double r = sqrt(x0 * x0 + x1 * x1);
+            double beta = k * r;
+
+            // Bessel functions
+            complex<double> Ho = jn(0, beta) + zi * yn(0, beta);
```

Comment:
There is inconsistent use of `std::complex<double>` and `complex<double>`. You could just use the latter everywhere.

Reply:
fixed